src/pydap/net.py

```diff
@@ -3,15 +3,18 @@
 
 from six.moves.urllib.parse import urlsplit, urlunsplit
 
-def GET(url, application=None):
+
+def GET(url, application=None, session=None):
```

Comment:
It is verbose but the ``session`` object is carrying so much information and is so opposite to monkeypatching that I think it's worth the clutter. I think we are in agreement here.